src/build.sh

```diff
@@ -0,0 +1,403 @@
+        >&2 echo "Skipping build: libgmp already built for $target_arch"
+        return 0
+    fi
+
+    pushd "$gmp_build_dir" > /dev/null
+
+    ../configure --enable-static "CC=$CC" "CXX=$CXX" "--host=$HOST" 1>&2
+    if [[ $? -ne 0 ]]; then
+        return 1
+    fi
+
+    make -j$(nproc) 1>&2
+    if [[ $? -ne 0 ]]; then
+        return 1
+    fi
+
+    mkdir -p ./.libs/include/
+    cp gmp.h ./.libs/include/
+    mkdir -p ./.libs/lib/
+    cp ./.libs/libgmp.a ./.libs/lib/
+
+    popd > /dev/null
+}
+
+function build_libmpfr() {
+    # Build libmpfr.
+    #
+    # Parameters:
+    # $1: mpfr package directory
+    # $2: libgmp build directory
+    # $3: target architecture
+    #
+    # Echoes:
+    # The libmpfr build directory
+    #
+    # Returns:
+    # 0: success
+    # 1: failure
+
+    local mpfr_dir="$1"
+    local libgmp_build_dir="$2"
+    local target_arch="$3"
+
+    local mpfr_build_dir="$(realpath "$mpfr_dir/build-$target_arch")"
+    mkdir -p "$mpfr_build_dir"
+    echo "$mpfr_build_dir"
+
+    if [[ -f "$mpfr_build_dir/src/.libs/lib/libmpfr.a" ]]; then
+        >&2 echo "Skipping build: libmpfr already built for $target_arch"
+        return 0
+    fi
+
+    pushd "$mpfr_dir/build-$target_arch" > /dev/null
+
+    ../configure --enable-static "--with-gmp-build=$libgmp_build_dir" "CC=$CC" "CXX=$CXX" "--host=$HOST" 1>&2
+    if [[ $? -ne 0 ]]; then
+        return 1
+    fi
+
+    make -j$(nproc) 1>&2
+    if [[ $? -ne 0 ]]; then
+        return 1
+    fi
+
+    mkdir -p ./src/.libs/include
+    cp ../src/mpfr.h ./src/.libs/include/
+    mkdir -p ./src/.libs/lib
+    cp ./src/.libs/libmpfr.a ./src/.libs/lib/
+
+    popd > /dev/null
+}
+
+function build_gdb() {
+    # Configure and build gdb.
+    #
+    # Parameters:
+    # $1: gdb directory
+    # $2: target architecture
+    # $3: libiconv prefix
+    # $4: libgmp prefix
+    # $5: libmpfr prefix
+    #
+    # Echoes:
+    # The gdb build directory
+    #
+    # Returns:
+    # 0: success
+    # 1: failure
+
+    local gdb_dir="$1"
+    local target_arch="$2"
+    local libiconv_prefix="$3"
+    local libgmp_prefix="$4"
+    local libmpfr_prefix="$5"
+
+    local gdb_build_dir="$(realpath "$gdb_dir/build-$target_arch")"
+    echo "$gdb_build_dir"
+    mkdir -p "$gdb_build_dir"
+
+    if [[ -f "$gdb_build_dir/gdb/gdb" ]]; then
+        >&2 echo "Skipping build: gdb already built for $target_arch"
+        return 0
+    fi
+
+    pushd "$gdb_build_dir" > /dev/null
+
+    ../configure --enable-static --with-static-standard-libraries --disable-tui --disable-inprocess-agent \
+                 "--with-libiconv-prefix=$libiconv_prefix" --with-libiconv-type=static \
+                 "--with-gmp=$libgmp_prefix" \
+                 "--with-mpfr=$libmpfr_prefix" \
+                 "CC=$CC" "CXX=$CXX" "--host=$HOST" 1>&2
+    if [[ $? -ne 0 ]]; then
+        return 1
+    fi
+
+    make -j$(nproc) 1>&2
+    if [[ $? -ne 0 ]]; then
+        return 1
+    fi
+
+    "$STRIP" ./gdb/gdb
```

Comment:
Something about this bugs me
The fact that we need to strip manually may mean that we are compiling the packages in a "debug" mode. Could you check that this is not the case? (it might make gdb a little bit slower because of that)

Reply:
Well, I tried looking into that.... and sadly, didn't find anything.
I looked into the building systems of [Fedora](https://kojipkgs.fedoraproject.org//packages/gdb/15.1/2.eln142/data/logs/x86_64/build.log) and of [Ubuntu](https://launchpadlibrarian.net/748664403/buildlog_ubuntu-oracular-amd64.gdb_15.1-1ubuntu2_BUILDING.txt.gz).
It seems that Ubuntu is using `dh_strip`, and I don't even know what the Fedora guys are doing.

I know that it's sad, but I don't see a better option for now.

Reply:
Hmmmm. could you make sure gdb is built with optimizations? Sorry for troubling you so much :pray:

Reply:
Sure! That is a great idea!

Reply:
Well, foolish me. You were right!
It seems that compiling the binary with -O2 makes it drop its size significantly!

Reply:
Hmmm, also make sure the libraries are optimized as well :)